### PR TITLE
[FIX] evaluation: compute cell only once

### PR DESCRIPTION
**Pull request by LucasLefevre**
## Description:

At the first evaluation, cells are computed in the order they are defined. When a formula references another cell, the cell is computed (if it's not already computed). However, the result is currently not stored. Therefore, if another formula references the same cell, it will be recomputed!

It can drastically slow down the evaluation as the same cells can be recomputed an enormous amount of time.

In a (quite big) spreadsheet prepared for a demo at OXP, evaluation goes from more than 30 seconds to less than 500ms

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/plugins/ui_core_views/cell_evaluation/evaluator.ts [hunk 1]
```diff
@@ -386,6 +382,17 @@ export class Evaluator {
...
   }
 
+  private buildCompilationParameters() {
+    return buildCompilationParameters(this.context, this.getters, (position) => {
+      const positionId = this.encodePosition(position);
+      const evaluatedCell = this.computeCell(positionId);
+      if (!this.evaluatedCells.has(positionId)) {
+        this.setEvaluatedCell(positionId, evaluatedCell);
+      }
+      return evaluatedCell;
```
Comment (LucasLefevre) [inline src/plugins/ui_core_views/cell_evaluation/evaluator.ts:392]: I'd like to extract the callback to a method, but I don't know how to name it :/

Comment (laa-odoo) [reply src/plugins/ui_core_views/cell_evaluation/evaluator.ts:392]: computeAndSave ?

**Review by laa-odoo — APPROVED** [review]

**Comment by laa-odoo** [comment]
@robodoo r+

**Comment by LucasLefevre** [comment]
robodoo rebase-ff

**Pull request closed without merging**